### PR TITLE
[Merged by Bors] - TO-3325 save interacted documents

### PR DESCRIPTION
**Pull request by janpetschexain**
**References**

- [TO-3325]
- see #615

**Summary**

- store interacted document and user ids in a table via a n:n relation

[TO-3325]: https://xainag.atlassian.net/browse/TO-3325?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by janpetschexain** [comment]
@biernacki i had to reopen the pr because github was to eager to close the previous one :/

File: discovery_engine_core/web-api/src/storage.rs [hunk 1]
```diff
@@ -163,11 +177,19 @@ impl UserState {
...
+            .execute(&mut tx)
+            .await?
+            .rows_affected()
+            > 0
+            || deletion;
 
         tx.commit().await?;
 
-        Ok(deletion.rows_affected() > 0)
+        Ok(deletion)
```
Comment (biernacki) [inline discovery_engine_core/web-api/src/storage.rs:192]: this will be removed by #621 so maybe don't add anything here to avoid conflicts

**Review by biernacki — COMMENTED** [review]
except changes to clear  handler which are redundant, the rest looks good

**Review by biernacki — APPROVED** [review]
lgtm, nice work 👍

**Comment by janpetschexain** [comment]
bors merge

**Pull request closed without merging**